components/brave_wallet_ui/common/slices/endpoints/coingecko-endpoints.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+import {
+  BraveWallet
+} from '../../../constants/types'
+import { WalletApiEndpointBuilderParams } from '../api-base.slice'
+
+export const coingeckoEndpoints = ({
+  mutation,
+  query
+}: WalletApiEndpointBuilderParams) => {
+  return {
+    getCoingeckoId: query<
+      string | undefined,
+      Pick<BraveWallet.BlockchainToken, 'chainId' | 'contractAddress'>
+    >({
+      queryFn: async (
```

Comment:
please wrap in a try-catch

Reply:
https://github.com/brave/brave-core/pull/19489/commits/7c0c9c2fe38d7abcae6ad041944bfc17095daa45